From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Outfluencer <48880402+Outfluencer@users.noreply.github.com>
Date: Mon, 26 Dec 2022 09:05:00 +0200
Subject: [PATCH] Reduce use of ServerData packet

Avoids the popup message box "Chat messages can't be verified"

diff --git a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
index e83fa08da12a9542fa1d21267275e53cc286a640..3be3417372e3a605c8c396f8fff55cfd610f81a1 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java
@@ -89,6 +89,7 @@ public class DownstreamBridge extends PacketHandler
     private final ProxyServer bungee;
     private final UserConnection con;
     private final ServerConnection server;
+    private boolean enforceSecure; // IvanCord
 
     @Override
     public void exception(Throwable t) throws Exception
@@ -1034,9 +1035,11 @@ public class DownstreamBridge extends PacketHandler
     @Override
     public void handle(ServerData serverData) throws Exception
     {
+        if ( enforceSecure ^ ( enforceSecure = serverData.isEnforceSecure() ) ) { // IvanCord
         serverData.setMotd( null );
         serverData.setIcon( null );
         con.unsafe().sendPacket( serverData );
+        } // IvanCord
         throw CancelSendSignal.INSTANCE;
     }
 
